MIRI_NO_STD can lead to very confusing errors

To reproduce, consider the following crate:
```
[package]
name = "embeddedMiri"
version = "0.1.0"
edition = "2021"

[profile.dev]
panic ='abort'

[profile.release]
panic ='abort'

[dependencies]
panic-semihosting = "0.6.0"
```
with `main.rs` being
```rust
#![no_std]
#![feature(start)]

use panic_semihosting as _;

#[start]
fn main(_argc: isize, _argv: *const *const u8) ->isize{
    panic!("hello")
}
```
It turns out `panic_semihosting` actually needs `std` on targets that do not satisfy `all(target_arch = "arm", target_os = "none")`.

When running this with a truly no-std target such as  `x86_64-unknown-none`, this is very clear from the error:
```rust
error[E0463]: can't find crate for `std`
  |
  = note: the `x86_64-unknown-none` target may not support the standard library
  = note: `std` is required by `panic_semihosting` because it does not declare `#![no_std]`
  = help: consider building the standard library from source with `cargo build -Zbuild-std`
```

However, when one does `MIRI_NO_STD=1 cargo miri run` to run this without std on a target that has std, the error is pretty bad:
```
error[E0463]: can't find crate for `panic_semihosting`
 --> src/main.rs:4:5
  |
4 | use panic_semihosting as _;
  |     ^^^^^^^^^^^^^^^^^ can't find crate

error[E0463]: can't find crate for `panic_semihosting`
 --> src/main.rs:4:5
  |
4 | use panic_semihosting as _;
  |     ^^^^^^^^^^^^^^^^^ can't find crate
  |
  = note: duplicate diagnostic emitted due to `-Z deduplicate-diagnostics=no`

error: `#[panic_handler]` function required, but not found

error: aborting due to 3 previous errors
```

I don't know exactly why that is. But my current inclination is to just remove the MIRI_NO_STD env var; it shouldn't be needd as `-none` targets get built without std automatically...

--- a/compiler/rustc_metadata/src/errors.rs
+++ b/compiler/rustc_metadata/src/errors.rs
@@ -5,7 +5,6 @@ use std::{
 
 use rustc_errors::{codes::*, Diag, DiagCtxt, Diagnostic, EmissionGuarantee, Level};
 use rustc_macros::{Diagnostic, Subdiagnostic};
-use rustc_session::config;
 use rustc_span::{sym, Span, Symbol};
 use rustc_target::spec::{PanicStrategy, TargetTriple};
 
@@ -640,9 +639,7 @@ impl<G: EmissionGuarantee> Diagnostic<'_, G> for CannotFindCrate {
         diag.arg("locator_triple", self.locator_triple.triple());
         diag.code(E0463);
         diag.span(self.span);
-        if (self.crate_name == sym::std || self.crate_name == sym::core)
-            && self.locator_triple != TargetTriple::from_triple(config::host_triple())
-        {
+        if self.crate_name == sym::std || self.crate_name == sym::core {
             if self.missing_core {
                 diag.note(fluent::metadata_target_not_installed);
             } else {

